### PR TITLE
Add a working basic loading and showing products

### PR DESCRIPTION
**Pull request by depop-james-hughes**

### CODE REVIEW

File: depop-app/src/components/ProductsContainer.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import "./ProductsContainer.css"
+
+export const ProductsContainer = () => {
+
+    const {products, loading} = useGetProducts('https://5c78274f6810ec00148d0ff1.mockapi.io/api/v1/products');
+
+    return (
+        <React.Fragment>
+            {!loading && products &&
+                <div className={"subTitle"}><b>{products.length} Results</b></div>
```
Comment (timnovis) [inline depop-app/src/components/ProductsContainer.js]: Let's leave this out and leave it for candidates to implement (often interesting to see how they handle this behaviour once products are filtered)

File: depop-app/src/components/ProductsContainer.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            }
+            {loading && <span>Loading items</span>}
```
Comment (timnovis) [inline depop-app/src/components/ProductsContainer.js]: D'ya reckon we should remove this to see if they implement it?

File: depop-app/src/hooks/useGetProducts.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import {useEffect, useState} from "react";
+
+export const useGetProducts = (url) => {
+    const [products, setProducts] = useState(null);
```
Comment (timnovis) [inline depop-app/src/hooks/useGetProducts.js]: I'd probs rec initialising this as an empty array over null

File: depop-app/src/hooks/useGetProducts.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import {useEffect, useState} from "react";
+
+export const useGetProducts = (url) => {
```
Comment (timnovis) [inline depop-app/src/hooks/useGetProducts.js:3]: I can't tell if this is too much to include or not, I'd be interested to know what others think as it is interesting to see different candidates approaches to data fetching

File: depop-app/src/components/ProductCard.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import "./ProductCard.css"
+
+export const ProductCard = ({product}) => {
+    console.log(product);
```
Comment (timnovis) [inline depop-app/src/components/ProductCard.js]: Remove?

File: depop-app/src/App.js [hunk 6]
```diff
@@ -1,25 +1,34 @@
...
+                    </path>
+                    </svg>
+                </span>
+                <p className={"title"}>
```
Comment (timnovis) [inline depop-app/src/App.js]: No need for {}

File: depop-app/src/App.js [hunk 7]
```diff
@@ -1,25 +1,34 @@
...
+                    <a
+                        href="https://github.com/depoplabs/web-pair-programming-assessment/blob/master/assessment.md"
+                        target="_blank"
+                        rel="noopener noreferrer">
+                        Link to task instructions
+                    </a>
+                </p>
+
+            </header>
+            <div className={"container"}>
```
Comment (timnovis) [inline depop-app/src/App.js]: No need for {}

**Review by timnovis — COMMENTED** [review]
Nice one - have left a few comments, whole thing needs prettier and eslint running over it though as the formatting in places is a bit off!

File: depop-app/src/components/ProductCard.js (hunk 5 above)
Comment (depop-james-hughes) [reply depop-app/src/components/ProductCard.js]: I was thinking of keeping it in, so candidates could quickly see the response from the endpoint - but I guess then can just call the API in their browser.

File: depop-app/src/components/ProductsContainer.js (hunk 2 above)
Comment (depop-james-hughes) [reply depop-app/src/components/ProductsContainer.js]: Good question. I guess it's a pretty one to quickly implement to get someone building confidence. I could leave the variable in the hook, but just not import it yet.

File: depop-app/src/components/ProductsContainer.js (hunk 1 above)
Comment (depop-james-hughes) [reply depop-app/src/components/ProductsContainer.js]: Sure.

**Review by timnovis — APPROVED** [review]

**Pull request merged**